drills/version-3/drill3reactWithRedux/src/App.js

```diff
@@ -1,72 +1,84 @@
-import React, { Component } from 'react';
-import Header from './Components/Header';
-import JobDetails from './Components/JobDetails';
-import Form from './Components/Form';
-import Preview from './Components/Preview';
-import Footer from './Components/Footer';
-import './App.css';
-import { connect } from 'react-redux';
-import * as actionTypes from './store/actions'
+import React, { Component } from "react";
+import Header from "./Components/Header";
+import JobDetails from "./Components/JobDetails";
+import Form from "./Components/Form";
+import Preview from "./Components/Preview";
+import Footer from "./Components/Footer";
+import "./App.css";
+import { connect } from "react-redux";
+import * as actionTypes from "./store/actions";
 
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
       title: [],
-      description: '',
+      description: ""
     };
   }
 
   componentDidMount() {
-    let apiURL = './listing.json';
+    let apiURL = "./listing.json";
     fetch(apiURL)
-    .then(response => response.json())
-    .then(data => {
-      this.setState({
-        title: data.title,
-        description: data.description
-      })
-    })
+      .then(response => response.json())
+      .then(data => {
+        this.setState({
+          title: data.title,
+          description: data.description
+        });
+      });
   }
-  
+
+  onKeyUp = event => {
```

Comment:
These two methods do whatever you need to with the event, and then just pass data to a redux action.